### ISSUE TITLE
Is there a way to prevent some task to run on thread 0 ?

### ISSUE THREAD

**Issue by loudjanilef** [dougbinks/enkiTS#130]
Hello,
First of all, thank you for providing this very good library.

I have a project with a main thread (enki thread 0) and N enki threads.
In this project, we have some high priority background tasks we want to schedule only on the workers threads and not on the main thread. This is because those tasks are long and unsliceable. 
We don't want those task to be scheduled on the main thread when we are waiting for other tasks. 
Waiting for a priority is not an option either since those tasks need to be scheduled as soon as possible (on any other thread) so they have the higher priority.
PinnedTasks does not provide the api we want since it will lock this high priority task on a given thread even if it is not available.

Do you have a way to set task <-> worker thread affinity or exclusion list ?

**Comment by dougbinks** [comment]
> Do you have a way to set task <-> worker thread affinity or exclusion list ?

No, the only current exclusion list capability is the priority or pinned tasks.

I use the priority system for this. Define a set of priorities for the main thread (and other threads), and a set below this for only other threads.

Some form of task <> thread affinity is a reasonable ask, but it's tricky to implement (there are some edge cases) with enkiTS without a fair amount of extra functionality.

**Comment by loudjanilef** [comment]
Thank you
I'll try with priorities, although it is not the best way because I would like the "non main thread" tasks to have a high priority too.
A thread affinity functionality would be very appreciate, but I understand that this can bring complexity.

**Comment by eugeneko** [comment]
@dougbinks I noticed "similar but not quite" issue.

If `WriterTryWriteFront` fails, `SplitAndAddTask` may execute task instead of adding it, which may be unacceptable if I am posting heavy task from low-latency thread. I would like to have a guarantee that no matter what happens, added task is not executed during  adding.

This issue is similar in title, but the content is different. I don't really need thread affinities or whatever. I just want "AddTask" to only add task and not execute it.

I think the problem is that task pipe has finite and quite small constant size. Can this value be configurable?

**Comment by dougbinks** [comment]
@eugeneko this is a different issue, please see the comment I have added in #62
